src/testdrive/src/action.rs

```diff
@@ -347,39 +347,10 @@ impl State {
         F: FnOnce(ConnCatalog) -> T,
     {
         if let Some(url) = &self.materialize_catalog_postgres_stash {
-            let schema = format!("mz_stash_copy_{}", self.seed);
-
-            let (client, _) = postgres_client(url).await?;
-
-            let current_schema: String =
-                client.query_one("SELECT current_schema", &[]).await?.get(0);
-
-            client
-                .execute(format!("CREATE SCHEMA {schema}").as_str(), &[])
-                .await?;
-            client
-                .execute(format!("SET search_path TO {schema}").as_str(), &[])
-                .await?;
-            client
-                .batch_execute(&format!(
-                    "
-                    CREATE TABLE fence AS SELECT * FROM {0}.fence;
-                    CREATE TABLE collections AS SELECT * FROM {0}.collections;
-                    CREATE TABLE data AS SELECT * FROM {0}.data;
-                    CREATE TABLE sinces AS SELECT * FROM {0}.sinces;
-                    CREATE TABLE uppers AS SELECT * FROM {0}.uppers;
-                    ",
-                    current_schema,
-                ))
-                .await?;
```

Comment:
Is all this schema and table stuff done in one of `make_tls`, `new_readonly`, or `open_debug`?

Reply:
Nope, the whole point of this PR is to no longer do any data copying. This changes the Catalog so that it can open from a readonly stash, which the caller of this function uses to compare the in memory catalog of a running environmentd to the on disk stash to ensure that all in memory changes are correctly persisted to disk.